tests/v2/validation/fleet/provisioning/airgap/fleet_in_airgap_test.go

```diff
@@ -0,0 +1,224 @@
+
+	a.fleetGitRepo = &v1alpha1.GitRepo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fleet.FleetMetaName + namegenerator.RandStringLower(5),
+			Namespace: fleet.Namespace,
+		},
+		Spec: v1alpha1.GitRepoSpec{
+			Repo:            fmt.Sprintf("%s@%s:/home/%s/%s", bastionUser, internalIP, bastionUser, fleetExampleFolderName),
+			Branch:          fleet.BranchName,
+			Paths:           []string{fleet.GitRepoPathLinux},
+			CorrectDrift:    &v1alpha1.CorrectDrift{},
+			ImageScanCommit: v1alpha1.CommitSpec{AuthorName: "", AuthorEmail: ""},
+			Targets: []v1alpha1.GitTarget{
+				{
+					ClusterSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      fleet.MatchKey,
+								Operator: fleet.MatchOperator,
+								Values: []string{
+									fleet.HarvesterName,
+								},
+							},
+						},
+					},
+				},
+			},
+			ClientSecretName: fleetSecretName,
+		},
+	}
+
+	a.client, err = a.client.ReLogin()
+	require.NoError(a.T(), err)
+
+	if a.clustersConfig.RKE2KubernetesVersions == nil {
+		rke2Versions, err := kubernetesversions.ListRKE2AllVersions(a.client)
+		require.NoError(a.T(), err)
+
+		a.clustersConfig.RKE2KubernetesVersions = []string{rke2Versions[len(rke2Versions)-1]}
+	}
+
+	if a.clustersConfig.CNIs == nil {
+		a.clustersConfig.CNIs = []string{fleet.CniCalico}
+	}
+}
+
+func (a *AirGapRKE2CustomClusterTestSuite) TestCustomClusterWithGitRepo() {
+	fleetVersion, err := fleet.GetDeploymentVersion(a.client, fleet.FleetControllerName, fleet.LocalName)
+	require.NoError(a.T(), err)
+	singleNodeRoles := []provisioninginput.MachinePools{provisioninginput.AllRolesMachinePool}
+
+	tests := []struct {
+		name        string
+		client      *rancher.Client
+		machinePool []provisioninginput.MachinePools
+	}{
+		{fleet.FleetName + " " + fleetVersion + "-" + permutations.RKE2AirgapCluster, a.standardClient, singleNodeRoles},
+	}
+	for _, tt := range tests {
+
+		testSession := session.NewSession()
+		defer testSession.Cleanup()
+
+		adminClient, err := a.client.WithSession(testSession)
+		require.NoError(a.T(), err)
+
+		provisioningConfig := *a.clustersConfig
+
+		provisioningConfig.Hardened = true
+		provisioningConfig.MachinePools = tt.machinePool
+
+		a.Run(tt.name, func() {
+
+			logrus.Info("Deploying airgap fleet gitRepo")
+			gitRepoObject, err := extensionsfleet.CreateFleetGitRepo(adminClient, a.fleetGitRepo)
+			require.NoError(a.T(), err)
+
+			logrus.Info("Deploying Custom Airgap Cluster")
+			testClusterConfig := clusters.ConvertConfigToClusterConfig(&provisioningConfig)
+
+			testClusterConfig.KubernetesVersion = a.clustersConfig.RKE2KubernetesVersions[0]
+			testClusterConfig.CNI = a.clustersConfig.CNIs[0]
+
+			clusterObject, err := provisioning.CreateProvisioningAirgapCustomCluster(tt.client, testClusterConfig, a.corralPackage)
+			require.NoError(a.T(), err)
+
+			reports.TimeoutClusterReport(clusterObject, err)
+			require.NoError(a.T(), err)
```

Comment:
reports.TimeoutClusterReport is not returning an error. This require.NoError here is still checking the error from CreateProvisioningAirgapCustomCluster, not from TimeoutClusterReport

Reply:
ack

Reply:
looks like the err is still from the previous object creation i.e., CreateProvisioningAirgapCustomCluster but not from the reports.TimeoutClusterReport.